datacube_stats/main.py

```diff
@@ -940,5 +965,168 @@ def __call__(self, product, time, group_by):
         return Tile(sources, geobox)
 
 
+class FilteredTileMaker(object):
```

Comment:
Could this share it's implementation with `ArbitraryTileMaker`, or even replace it.

I think we also want to be able to do the date filtering when not necessarily using polygons.